src/plans/components/ProviderSelect.tsx

```diff
@@ -0,0 +1,95 @@
+import { ComponentProps, FC, ReactNode, useMemo } from 'react';
+
+import Select from '@components/common/MtvSelect';
+import { EmptyState } from '@components/EmptyState';
+import { ProviderModelGroupVersionKind, ProviderModelRef, V1beta1Provider } from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  EmptyStateVariant,
+  Flex,
+  FlexItem,
+  Icon,
+  SelectOption,
+  Title,
+} from '@patternfly/react-core';
+import { isEmpty } from '@utils/helpers';
+import { getResourceUrl } from 'src/modules/Providers/utils/helpers/getResourceUrl';
+import { ExternalLink } from '@components/common/ExternalLink/ExternalLink';
+import { useForkliftTranslation } from '@utils/i18n';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+
+type ProviderSelectProps = Pick<ComponentProps<typeof Select>, 'onSelect' | 'status'> & {
+  id: string;
+  value: string;
+  namespace: string | undefined;
+  placeholder?: string;
+  emptyState?: ReactNode;
+};
+
+export const ProviderSelect: FC<ProviderSelectProps> = ({
+  id,
+  value,
+  namespace,
+  placeholder,
+  emptyState,
+  status,
+  onSelect,
+}) => {
+  const { t } = useForkliftTranslation();
+  const [providers, providersLoaded, providerError] = useK8sWatchResource<V1beta1Provider[]>({
+    namespace,
+    groupVersionKind: { ...ProviderModelGroupVersionKind, kind: 'what' },
+    namespaced: true,
+    isList: true,
+  });
+
+  const providersListUrl = useMemo(
+    () =>
+      getResourceUrl({
+        reference: ProviderModelRef,
+        namespace: namespace,
+      }),
+    [namespace],
+  );
+
+  return (
+    <Select
+      id={id}
+      value={value}
+      status={status}
+      onSelect={onSelect}
+      placeholder={!providersLoaded && !providerError ? t('Loading...') : placeholder}
+    >
+      {providerError && providerError instanceof Error ? (
```

Comment:
why checking instanceOf? do we need it? the error type coming from SDK hok, this type can be changed.. i don't thing checking instanceOf is needed

Reply:
providerError has no type. I looked at the tests for `useK8sWatchResource`, and they test it out with a thrown Error type, but technically the type is `any`. We only want to read a message from an error object if what is returned is an instance of an Error, otherwise, we're guessing, no?

Reply:
The typeahead selects that fetch namespaces don't have this dropdown option yet to show API error messages, would you like me to add that?

Reply:
This line was removed and the error dropdown in general will be handled separately; https://issues.redhat.com/browse/MTV-2347